packages/studio/src/components/AddPageButton/EntityIdField.tsx

```diff
@@ -0,0 +1,140 @@
+import Select, { MultiValue, StylesConfig } from "react-select";
+import useStudioStore from "../../store/useStudioStore";
+import StreamScopeFieldLabel from "./StreamScopeFieldLabel";
+import { ChangeEvent, useCallback, useMemo, useState } from "react";
+import { pillContainerClass } from "../PillPicker/PillPickerInput";
+import classNames from "classnames";
+
+interface Props {
+  disabled: boolean;
+  updateSelection: (selectedIds: string[]) => void;
+  selectedIds?: string[];
+}
+
+export default function EntityIdField({
+  disabled,
+  updateSelection,
+  selectedIds,
+}: Props) {
+  const [entitiesRecord, fetchEntities] = useStudioStore((store) => [
+    store.accountContent.entitiesRecord,
+    store.accountContent.fetchEntities,
+  ]);
+  const availableEntityTypes = disabled ? [] : Object.keys(entitiesRecord);
+  const [isLoading, setIsLoading] = useState(false);
+  const [entityType, setEntityType] = useState<string | undefined>(
+    availableEntityTypes[0]
+  );
+  const entityOptions = useEntityOptions(entityType);
+  const selectedOptions = useMemo(
+    () => selectedIds?.map((id) => ({ value: id, label: id })),
+    [selectedIds]
+  );
+
+  const onEntityTypeChange = useCallback(
+    (e: ChangeEvent<HTMLSelectElement>) => {
+      setEntityType(e.target.value);
+    },
+    []
+  );
+
+  const onEntityIdChange = useCallback(
+    (selectedItems: MultiValue<{ value: string }>) => {
+      updateSelection(selectedItems.map((item) => item.value));
+    },
+    [updateSelection]
+  );
+
+  const onMenuScrollToBottom = useCallback(async () => {
+    if (!entityType) {
+      return;
+    }
+    const currentNumEntities = entitiesRecord[entityType].entities.length;
+    if (currentNumEntities >= entitiesRecord[entityType].totalCount) {
+      return;
+    }
+    setIsLoading(true);
+    await fetchEntities(entityType, Math.floor(currentNumEntities / 50));
+    setIsLoading(false);
+  }, [entitiesRecord, entityType, fetchEntities]);
+
+  const containerClass = "flex flex-col mb-4";
+
+  if (availableEntityTypes.length === 0) {
+    return (
+      <div className={containerClass}>
+        <StreamScopeFieldLabel streamScopeField="entityIds" />
+        <div
+          className={classNames(
+            pillContainerClass,
+            "mt-2 bg-gray-50 text-gray-500 pb-2"
+          )}
+        >
+          {!disabled && "No entity types found in the account."}
```

Comment:
nit: this should be "No entities found in the account."

Reply:
Oh hmm this message appears when availableEntityTypes has a length of 0, but I guess that no entity types also means no entities at all? Anyways I'll update the message